src/cli.c

```diff
@@ -1645,32 +1715,66 @@ get_config_key_result(char* config_key, struct json* j, uintptr_t* r, int32_t ou
    if (output_format == MANAGEMENT_OUTPUT_FORMAT_JSON || !config_key)
    {
       *r = (uintptr_t)filtered_response;
+
       free(config_value);
+      config_value = NULL;
+
    }
    else
    {
       *r = (uintptr_t)config_value;
       pgagroal_json_destroy(filtered_response);
    }
 
+   // Clean up parts
+   for (int i = 0; i < part_count && i < 4; i++)
+   {
+      free(parts[i]);
+      parts[i] = NULL;
+   }
+
+   free(config_key_copy);
+   config_key_copy = NULL;
+
    return 0;
 
 error:
-
    if (output_format == MANAGEMENT_OUTPUT_FORMAT_JSON)
    {
       pgagroal_json_put(filtered_response, "Outcome", (uintptr_t)false, ValueBool);
       *r = (uintptr_t)filtered_response;
+
       free(config_value);
+      config_value = NULL;
+
    }
    else
    {
+
+      free(config_value);
+      config_value = NULL;
+
       config_value = (char*)malloc(6);
-      memcpy(config_value, "Error\0", 6);
+      if (config_value)
+      {
+         snprintf(config_value, 6, "Error");
+      }
       *r = (uintptr_t)config_value;
       pgagroal_json_destroy(filtered_response);
    }
 
+   // Clean up parts on error
+   for (int i = 0; i < part_count && i < 4; i++)
```

Comment:
same as above, could we check in the beginning that `part_count` is  at max `3`?